guacamole-ext/src/main/java/org/apache/guacamole/token/TokenFilter.java

```diff
@@ -247,19 +269,71 @@ public String filter(String input) {
 
             // Update last regex match
             endOfLastMatch = tokenMatcher.end();
-            
+
         }
 
         // Append any remaining non-token text
         output.append(input.substring(endOfLastMatch));
-        
+
         return output.toString();
-       
+
+    }
+
+    /**
+     * Filters the given string, replacing any tokens with their corresponding
+     * values. Any tokens present in the given string which lack values will
+     * be interpreted as literals.
+     *
+     * @param input
+     *     The string to filter.
+     *
+     * @return
+     *     A copy of the input string, with any tokens replaced with their
+     *     corresponding values.
+     */
+    public String filter(String input) {
```

Comment:
Why not roll this functionality into the 2-argiment `filter` method? Having another method with the same name that works _sort of_ but not _exactly like_ `filter(*, false)` is honestly a little confusing.

Reply:
Ah - because that would require all callers of `filter()` to handle the possible exception, even when there will not be one in the `false` case. There was some discussion on this in #336 (look for resolved conversations regarding `TokenFilter.java - linking to the conversation directly appears to not work for resolved discussions 🤷).

Reply:
Ahh right, gotcha. That makes sense.